### PR TITLE
ICU-20088 fix pages as part of userguide migration to markdown format.

### PR DESCRIPTION
**Pull request by gnrunge**
[ICU-20088] / #919 for the merge to master

# This is not a merge onto master.
DISABLE_JIRA_ISSUE_MATCH=true
ALLOW_MANY_COMMITS=true

[ICU-20088]: https://unicode-org.atlassian.net/browse/ICU-20088

### CODE REVIEW

**Comment by gnrunge** [comment]
Fixed.

File: docs/userguide/icudata.md [hunk 1]
```diff
@@ -53,21 +58,21 @@ to files located in the ICU data directory.
...
     built, use its compiled-in value.
 
 4.  Otherwise, the ICU data directory is an empty string. This is the default
     behavior for ICU using a shared library for its data and provides the
     highest data loading performance.
 
-*u_setDataDirectory() is not thread-safe. Call it before calling ICU APIs from
-multiple threads. If you use both u_setDataDirectory() and u_init(), then use
-u_setDataDirectory() first.*
+:point_right: **Note**: #`u_setDataDirectory()` is not thread-safe. Call it
```
Comment (markusicu) [inline docs/userguide/icudata.md]: The `#` doesn't belong, does it?

File: docs/userguide/icudata.md [hunk 2]
```diff
@@ -334,10 +351,10 @@ by their package name. Whenever a data item is requested from a package and that
 package has been loaded before, then the cached package is used immediately
 instead of searching through the filesystem.
 
-*ICU versions before 2.2 always searched data packages before looking for
-individual files, which made it impossible to override packaged data items. See
-the ICU 2.2 download page and the readme for more information about the
-changes.*
+> :point_right: *ICU versions before 2.2 always searched data packages before
```
Comment (markusicu) [inline docs/userguide/icudata.md]: not quite the agreed style -- `> :point_right: **Note**: that ICU for C by default ...`

(and please remove the italics)

File: docs/userguide/icudata.md [hunk 3]
```diff
@@ -375,9 +392,9 @@ If a directory separator (generally '/' or '\\') is needed in a path parameter,
 use the form that is native to the platform. The ICU header "putil.h" defines
 U_FILE_SEP_CHAR appropriately for the platform.
 
-*On Windows, the directory separator must be '\\' for any paths passed to ICU
-APIs. This is different from native Windows APIs, which generally allow either
-'/' or '\\'.*
+> :point_right: *On Windows, the directory separator must be '\\' for any paths
```
Comment (markusicu) [inline docs/userguide/icudata.md]: ditto

File: docs/userguide/icudata.md [hunk 4]
```diff
@@ -420,11 +439,11 @@ library is requested by using an option in the configure script at build time.
 Nothing is required at run time; ICU finds and uses whatever form of the data is
 available.
 
-*When the ICU data is built in the form of shared libraries, the library names
-have platform-specific prefixes and suffixes. On Unix-style platforms, all the
-libraries have the "lib" prefix and one of the usual (".dll", ".so", ".sl",
-etc.) suffixes. Other than these prefixes and suffixes, the library names are
-the same as the above .dat files.*
+> :point_right: *When the ICU data is built in the form of shared libraries, the
```
Comment (markusicu) [inline docs/userguide/icudata.md]: ditto

File: docs/userguide/icudata.md [hunk 5]
```diff
@@ -594,19 +572,174 @@ repository](https://github.com/unicode-org/icu/blob/master/icu4c/source/data/loc
...
 Data Repository](http://www.unicode.org/cldr/) ) project.
 
-As of ICU 64, dropping the txt file into the correct place in the source tree is
-sufficient to add it to your ICU build. You will need to re-configure in order
-to pick it up.
+**As of ICU 64**, dropping the txt file into the correct place in the source
+tree is sufficient to add it to your ICU build. You will need to re-configure in
+order to pick it up.
+
+**As of ICU 63 or earlier**,You need to write a resource bundle file for it
```
Comment (markusicu) [inline docs/userguide/icudata.md]: please use a heading (H2?) to set this off better and group its sub-sections

File: docs/userguide/icudata.md [hunk 6]
```diff
@@ -855,18 +994,19 @@ data, one of the resource files refers to the other file's data. In other cases,
 a file may alias a portion of another file's data in order to save space.
 Currently ICU4J provides no tool for revealing these dependencies.
 
-*Java's Locale class silently converts the language code "he" to "iw" when you
-construct the Locale (for versions of Java through Java 5). Thus Java cannot be
-used to locate resources that use the "he" language code. ICU, on the other
-hand, does not perform this conversion in ULocale, and instead uses aliasing in
-the locale data to represent the same set of data under different locale ids.*
+> :point_right: *Java's Locale class silently converts the language code "he" to
```
Comment (markusicu) [inline docs/userguide/icudata.md]: note style

File: docs/userguide/icudata.md [hunk 7]
```diff
@@ -879,14 +1019,14 @@ also references he_IL.res).
...
-want, and then create a new jar file with the remining resources. See the jar
+want, and then create a new jar file with the remaining resources. See the jar
 tool information for how to do this. Before 'rejaring' the files, be sure to
 thoroughly test your application with the remaining resources, making sure each
 required resource is present.
 
 #### Using additional resource files with ICU4J
 
-*Resource file formats can change across releases of ICU4J!*
+> :point_right: *Resource file formats can change across releases of ICU4J!*
```
Comment (markusicu) [inline docs/userguide/icudata.md]: note style

**Comment by gnrunge** [comment]
ll remarks worked in.

**Comment by gnrunge** [comment]
Should be: All remarks worked in.

**Pull request merged**